src/dimension_slice.c

```diff
@@ -142,42 +222,56 @@ dimension_slice_copy(const DimensionSlice *original)
 	return new;
 }
 
-static int
-cmp_slices_by_dimension_id(const void *left, const void *right)
+/*
+ * Check if two dimensions slices overlap by doing collision detection in one
+ * dimension.
+ *
+ * Returns true if the slices collide, otherwise false.
+ */
+bool
+dimension_slices_collide(DimensionSlice *slice1, DimensionSlice *slice2)
 {
-	const DimensionSlice *left_slice = *((DimensionSlice **) left);
-	const DimensionSlice *right_slice = *((DimensionSlice **) right);
-
-	if (left_slice->fd.dimension_id == right_slice->fd.dimension_id)
-		return 0;
-	if (left_slice->fd.dimension_id < right_slice->fd.dimension_id)
-		return -1;
-	return 1;
-}
-
+	Assert(slice1->fd.dimension_id == slice2->fd.dimension_id);
 
-static void
-hypercube_slice_sort(Hypercube *hc)
-{
-	qsort(hc->slices, hc->num_slices, sizeof(DimensionSlice *), cmp_slices_by_dimension_id);
+	return (slice1->fd.range_start < slice2->fd.range_end &&
```

Comment:
what about the case  (slice2->fd.range_start < slice1->fd.range_end &&
slice2->fd.range_end > slice1->fd.range_start); Otherwise, if we expect ordered input, I think there should be an additional Assert.

Reply:
What you are describing is the same case if you look closely. It doesn't matter which slice you have first in the expression. Either they collide or they do not.